### ISSUE TITLE
Network.image force retry if failed

### ISSUE THREAD

**Issue by peterlauri** [flutter/flutter#55364]
Steps to reproduce:

1. Create two pages separate screens, screen1 contains a link to screen2, and screen2 contains a Network.image with a valid resource
2. Turn off internet connection and navigate to screen2, image fails to load and no image is shown
3. Go back to screen1 and turn on internet connection
4. Navigate to screen2

Expected: image is loaded
Actual: image is not loaded

Guess this is due to how the caching is implemented. But is there away around this using Image.network?

**Comment by Abhishek01039** [comment]
can you send code?

**Comment by VladyslavBondarenko** [comment]
Hi @peterlauri 
provide please your `flutter doctor -v `.
I tried this code sample with `dev v1.18.0-6.0.pre` on iOS 13.4.5. Couldn't reproduce, Image is loaded on the step 4.

```
import 'package:flutter/material.dart';

void main() {
  runApp(MaterialApp(
    title: 'Navigation Basics',
    home: FirstRoute(),
  ));
}

class FirstRoute extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('First Route'),
      ),
      body: Center(
        child: RaisedButton(
          child: Text('Open route'),
          onPressed: () {
            Navigator.push(
              context,
              MaterialPageRoute(builder: (context) => SecondRoute()),
            );
          },
        ),
      ),
    );
  }
}

class SecondRoute extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Second Route"),
      ),
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: <Widget>[
          Image.network('https://flutter.dev/images/flutter-logo-sharing.png'),
          RaisedButton(
            onPressed: () {
              Navigator.pop(context);
            },
            child: Text('Go back!'),
          ),
        ],
      ),
    );
  }
}
```

**Comment by peterlauri** [comment]
https://github.com/peterlauri/flutter_image_network_problem

This is how screen2 should look like:

![image](https://user-images.githubusercontent.com/3514945/79972975-6fe27500-849f-11ea-86bd-83c4c39c61d5.png)

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/3514945/79972955-678a3a00-849f-11ea-99af-9223bcb5a785.gif)

**Comment by peterlauri** [comment]
@VladyslavBondarenko just added my sample repo for this, and also a screen recording. For me it reproduces on iOS, web and Android. Beta channel.

```
➜  image_network_problem git:(master) flutter doctor -v
[✓] Flutter (Channel beta, v1.17.0, on Mac OS X 10.15.3 19D76, locale en-GB)
    • Flutter version 1.17.0 at /Users/plauri/.applications/flutter
    • Framework revision d3ed9ec945 (2 weeks ago), 2020-04-06 14:07:34 -0700
    • Engine revision c9506cb8e9
    • Dart version 2.8.0 (build 2.8.0-dev.18.0 eea9717938)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/plauri/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4.1, Build version 11E503a
    • CocoaPods version 1.9.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.44.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.9.1

[✓] Connected device (4 available)
    • PRA LX1             • 8NH7N17B04000425 • android-arm64  • Android 8.0.0 (API 26)
    • AOSP on IA Emulator • emulator-5554    • android-x86    • Android 9 (API 28) (emulator)
    • Chrome              • chrome           • web-javascript • Google Chrome 81.0.4044.113
    • Web Server          • web-server       • web-javascript • Flutter Tools

! Doctor found issues in 1 category.
```

**Comment by peterlauri** [comment]
@VladyslavBondarenko what channel are you on?

**Comment by VladyslavBondarenko** [comment]
@peterlauri 
current dev, checked with Android emulator - works as well

**Comment by peterlauri** [comment]
Channel stable, v1.12.13+hotfix.9 = **failed**
Channel beta, v1.17.0 = **failed**
Channel dev, v1.18.0-6.0.pre = **passed**

So this is fixed in dev channel. Was trying to find a related issue, but I cannot find it.

@VladyslavBondarenko do you know about related issue that is resolved? Want to find more info, and possible a workaround...

**Comment by VladyslavBondarenko** [comment]
@peterlauri 
no, sorry, I haven't seen and can't find an issue with resolving this case.
Closing as fixed, there is no need to track it
if you disagree please write in the comments and I will reopen it
Thank you

**Comment by peterlauri** [comment]
This looks related to https://github.com/flutter/flutter/issues/47270

The PR looks to be the one that solves this issue: https://github.com/flutter/flutter/pull/51398

However, in current Channel beta, v1.17.0 this still is an issue, but the commit 7cbe55cf1a looks to be included in beta.

@dnfield shouldn't this be solved by https://github.com/flutter/flutter/pull/51398 ?

**Comment by dnfield** [comment]
https://github.com/flutter/flutter/commit/30738982b40b4a7b2045377636531f7627450358 added some additional work related to this that I initially missed.  Can you try that commit?

**Comment by dnfield** [comment]
It's not available in v1.17.x

**Comment by peterlauri** [comment]
@dnfield how to properly test that commit? Should I just check out that commit in my flutter application directory? In my case /Users/plauri/.applications/flutter

**Comment by peterlauri** [comment]
I tried to just use the closest tagged version  1.18.0-dev.0.0, but there looks to be a bug, as it complains that it cannot find 1.18.0-de.0.0 (note the missing v in dev)...

```
➜  image_network_problem git:(master) ✗ flutter version 1.18.0-dev.0.0
╔══════════════════════════════════════════════════════════════════════════════╗
║ Warning: "flutter version" will leave the SDK in a detached HEAD state.      ║
║ If you are using the command to return to a previously installed SDK version ║
║ consider using the "flutter downgrade" command instead.                      ║
╚══════════════════════════════════════════════════════════════════════════════╝


Are you sure you want to proceed? [y|n]: y
There is no version: 1.18.0-de.0.0
Unable to checkout version branch for version 1.18.0-de.0.0: ProcessException: Process exited
abnormally:
error: pathspec '1.18.0-de.0.0' did not match any file(s) known to git
  Command: git checkout 1.18.0-de.0.0
```

**Comment by peterlauri** [comment]
@dnfield What I did:

- I switched to channel dev and ran flutter upgrade
- checked out the referenced commit 3073898 in my flutter repo
- flutter clean and launched the app

Result: success, so it looks like  3073898 solves this issue.

Will 1.17.x receive updates that are in dev? Or will beta not include those changes until beta is production ready?

**Comment by dnfield** [comment]
This fix is unlikely to make it in to v1.17.x, but over time beta (and stable) will roll to include this fix.

You are certainly free to cherry pick it into your local Flutter checkout if it's fixing things for you though :)
